source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -1,8 +1,236 @@
+   :language: go
+   :dedent:
+   :start-after: begin cursor try next
+   :end-before: end cursor try next
+
+.. note::
+   
+   If you previously used the cursor to iterate some documents, the
+   driver excludes those documents when you iterate again. 
+
+.. _cursor-elements-golang:
+
+Access Cursor Elements
+----------------------
+
+You can access a cursors elements before it's **exhausted**. A cursor is
+exhausted when the ``MongoClient`` iterates through its results and
+reaches its last element. Afterwards, the cursor won't respond to
+functions to access its elements.
+
+When :ref:`individually retrieving your cursors results
+<individual-documents-golang>`, you can view each document as a ``BSON`` or
+Go value.
+
+BSON Document
+~~~~~~~~~~~~~
+
+To access the current document as raw ``BSON``, use the ``Current``
+property.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin current
+   :end-before: end current
+
+Go Value Document
+~~~~~~~~~~~~~~~~~
+
+To access the current document as a Go value, use the ``Decode()``
+function. 
+
+This function unmarshals the current document into the :ref:`BSON
+interface <bson-types>` you specify. It returns any errors from the
+unmarshalling process without any modification.
+
+.. code-block:: go
+
+   var result bson.D
+   if err := cursor.Decode(&result); err != nil {
+      log.Fatal(err)
+   }
+   fmt.Println(result)
+
+Other Elements
+~~~~~~~~~~~~~~
+
+You can also retrieve the following elements:
+
+- `The number of documents remaining in the batch. <{+api+}/mongo#Cursor.RemainingBatchLength>`__
+- `The cursor ID. <{+api+}/mongo#Cursor.ID>`__ 
+- `The cursor errors encountered. <{+api+}/mongo#Cursor.Err>`__
+
+.. important:: Retrieving the Cursor ID
+
+   If you want to retrieve the cursor ID, include the option to set your
+   batch size. The batch size needs to be a number less than or equal to
+   the number of documents you expect to retrieve.
+   
+   .. code-block:: go
+      
+      opts := options.Find().SetBatchSize(<a number>)
+      cursor, err := coll.Find(context.TODO(), bson.D{}, opts)
+      if err != nil {
+         panic(err)
+      }
+
+   If you omit this option, set the batch size to a number greater than
+   the elements retrieved, or the ``MongoClient`` exhausts the cursor,
+   the cursor ID is ``0``.
+
+Close the Cursor
+----------------
+
+When your application no longer needs to use a cursor, close the cursor
+with the ``Close()`` method. This method free's up your cursor's
+consumption of resources in both the client application and the 
+MongoDB server.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin close
+   :end-before: end close
+
+.. note:: 
+
+   Close the cursor when you retrieve documents individually because
+   those functions make a cursor :manual:`tailable
+   </core/tailable-cursors/>`.
```

Comment:
Question/Suggestion: Are all cursors tailable? How do you open a tail-able/non-tailalbe cursor? The API docs seem to suggest that not all cursors are tailable? Could the mechanism to open a tailable vs non-tailable cursor be confirmed? If the driver contains both tailable and non-tailable cursors, could this note be clarified? 

```suggestion
.. note:: Close Tail-able Cursors 

   As :ref:`tail-able cursors <PRECEDING SECTION>` remain open indefinitely, make sure to manually close 
   tail-able cursors.
```